app/api/resources/resolver_view.py

```diff
@@ -1,49 +1,96 @@
+
   params = SearchPayload(
     stable_id = stable_id,
-    type = type
+    type = type,
+    per_page = 10
   )
+
+  # Get genome_ids from search api
   try:
     session = requests.Session()
-    search_results = {}
     with session.post(url=ENSEMBL_SEARCH_HUB_API,json=params.dict()) as response:
       response.raise_for_status()
       search_results = response.json()
-  
   except requests.exceptions.HTTPError as HTTPError:
     return response_error_handler({"status": HTTPError.response.status_code})
 
   except Exception as e:
     logger.exception(e)
     return response_error_handler({"status": 500})
-
-  if request.headers.get("content-type") == "application/json":
-    return search_results
   else:
     matches = search_results.get("matches")
-    if not matches:
-      return {"response": "No matches found"}
 
-    genome_id = matches[0].get("genome_id")
+  if not matches:
+    return response_error_handler({"status": 404})
 
-    if (app == "entity-viewer"):
-      url=f"{ENSEMBL_URL}/{app}/{genome_id}/{type}:{stable_id}"
+  # Get metadata for each genome
+  results = []
+  for match in matches:
+    genome_id = match.get("genome_id")
+    try:
+      with session.get(url=f"{ENSEMBL_URL}/api/metadata/genome/{genome_id}/details") as response:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as HTTPError:
+      return response_error_handler({"status": HTTPError.response.status_code})
+    except Exception as e:
+      logger.exception(e)
+      return response_error_handler({"status": 500})
     else:
-      url=f"{ENSEMBL_URL}/{app}/{genome_id}?focus={type}:{stable_id}"
+      meta_results = response.json()
+
+      if not meta_results:
+        return response_error_handler({"status": 404})
+
+      if app == "entity-viewer":
+        url = f"{ENSEMBL_URL}/{app}/{genome_id}/{type}:{stable_id}"
+      else:
+        url = f"{ENSEMBL_URL}/{app}/{genome_id}?focus={type}:{stable_id}"
+
+      meta = {
+        "accession_id": meta_results["assembly"]["accession_id"],
+        "assembly_name": meta_results["assembly"]["name"],
+        "species": meta_results["scientific_name"] if meta_results["scientific_name"] else meta_results["common_name"],
+        "type": meta_results["type"],
+        "resolved_url": url
+      }
+
+      results.append(meta)
+
+  # return request.headers
+
+  if "application/json" in request.headers.get("accept"):
+    return results
+
+  if len(results) == 1:
+    return RedirectResponse(results[0]["resolved_url"])
+  else:
+    return HTMLResponse(generate_html_content(results))
 
-    return RedirectResponse(url)
+def generate_html_content(results):
+    # Create a simple HTML page with a list of URLs
+  load_dotenv()
+  CURR_DIR = os.path.dirname(os.path.abspath(__file__))
+  env = Environment(loader=FileSystemLoader(os.path.join(CURR_DIR,"templates")))
+  interstitial_template = env.get_template("interstitial.j2")
```

Comment:
In examples in FastAPI docs, they use the html extension for jinja templates. Do we prefer a j2 extension?

Reply:
Although we used .j2 in [ticket manager](https://gitlab.ebi.ac.uk/ensembl-web/ensembl_ticket_manager/-/tree/master/app/api/resources/mail_manager/templates?ref_type=heads)
I prefer using .html. If thats not a problem then I will update this. @Mehrnaz-Charkhchi @kamaldodiya ?

Reply:
@jyothishnt It doesn't seem to be making a difference according to the docs so I would say go with whatever you're comfortable with.